scripts/benchmark_ipa.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# Generate stage-wise report after running IPA bench
+#
+# To use, supply the number of rows you want to run benchmarking on
+
+num_rows=$1
+if [ "$#" -ne 1 ]; then
+    echo "Please supply query size as an argument to the script"
+    exit
+fi
+
+echo "Step,Records,Bytes"
+
+DISPLAY_STAGE=("Verify" "Sort" "Attribution" "Capping" "Aggregation" "Others")
+
+records=
+bytes=
+
+for i in "${DISPLAY_STAGE[@]}"; do 
+  records+=(0); 
+  bytes+=(0);
+done
+
+while IFS=, read -d ' ' actual_step_name a b c d; do
+    if [[ $actual_step_name != *"protocol"* ]]; then 
+      continue
+    fi
+
+    # In case you add a new stage to DISPLAY_STAGE, please add corresponding step 
+    # in below case to match expected DISPLAY_STAGE array index
```

Comment:
You could maybe remove maintenance overhead with some meta-programming, but this is fine for now...

```bash
steps() {
if [[ "$1" == "f" ]]; then
    echo 'match_step() {'
    echo '    case "$1" in'
fi
i=0
while read name pattern; do
    if [[ "$1" == "f" ]]; then
        echo "        ${pattern}) step=$i ;;"
    else
        echo "DISPLAY_STAGE+=(\"$name\");"
    fi
    i=$((i + 1))
done <<EOSTEPS
Verify */mod_conv_breakdown_key/*|*/mod_conv_match_key/*
Sort */apply_sort_permutation/*|*/gen_sort_permutation_from_match_keys/*
EOSTEPS
if [[ "$1" == "f" ]]; then
   echo '    esac'
    echo '}'
fi
}
eval "$(steps f)"
DISPLAY_STAGE=()
eval "$(steps n)"
```

Reply:
This makes the script very hard to read :p can we leave it as is for now?